cmd/router/main_test.go

```diff
@@ -515,3 +515,53 @@ func TestCallServiceWhenMultipleHeadersToPropagate(t *testing.T) {
 	fmt.Printf("final response:%v\n", response)
 	assert.Equal(t, expectedResponse, response)
 }
+
+func TestModelSwitchExample(t *testing.T) {
+	// Start port-forwarding to istio-ingressgateway
+	portForwardCmd := exec.Command("kubectl", "port-forward", "svc/istio-ingressgateway", "-n", "istio-system", "8080:80")
```

Comment:
I guess this assumes that istio-ingressgateway is there. It is not a unit test rather an integration test. For unit test, you should be able to mock external dependencies.

You just need to test that when callService tries to call a malformed URL then it throws an err that you handled now.

See this unit test [TestCallServiceWhen1HeaderToPropagate](https://github.com/kserve/kserve/blob/68430f9ad8c8848c62520fe9b3028c0158b9227f/cmd/router/main_test.go#L426) for reference.

Reply:
@rachitchauhan43 I added unit tests and removed integration tests, and I also fixed the DCO check.